sys/kern/kern_physio.c

```diff
@@ -173,10 +174,17 @@ physio(struct cdev *dev, struct uio *uio, int ioflag)
 			bp->bio_dev = dev;
 
 			if (pages) {
+#if __has_feature(capabilities) && !defined(__CHERI_PURE_CAPABILITY__)
+				/* XXX: bp->bio_data should be a cap */
```

Comment:
Actually, I don't think so.  This is the _only_ place in the entire kernel  that stores a user pointer in bio_data and it does it for about 20 lines of C.  I think instead we should use the same logic I just applied in vfs_aio.c where we use a temporary variable for the user address throughout this function and only assign kernel addresses to `bio_data` here and in the other two cases below.  That is, I would not set `bio_data` up above but instead use an else here for the `pages == NULL` case to set `bio_data` to `iov_base`.

Reply:
I've fixed this one, will include in my next push.